cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -625,7 +625,8 @@ private float getTruncatedLowerBounds(Expr expr) {
               )
           else result = newLB
         else result = exprMinVal(expr)
-      )
+      ) and
+      getUpperBoundsImpl(expr) <= exprMaxVal(expr)
```

Comment:
Do we need a similar conjunct in `getTruncatedUpperBounds`?

Reply:
In principle yes, but in practice we may not need it because positive overflow is more common than negative overflow. Adding this conjunct introduces non-linear recursion, which is likely to make things slower on non-pathological databases.

When CPP-Differences finishes, the trade-off will hopefully become clearer.